integration-libs/opf/base/core/facade/opf-payment.service.ts

```diff
@@ -24,18 +53,130 @@ export class OpfPaymentService implements OpfPaymentFacade {
     },
     OpfPaymentVerificationResponse
   > = this.commandService.create((payload) =>
-    this.opfCheckoutConnector.verifyPayment(
+    this.opfPaymentConnector.verifyPayment(
       payload.paymentSessionId,
       payload.paymentVerificationPayload
     )
   );
 
+  protected submitPaymentCommand: Command<
+    {
+      submitInput: SubmitInput;
+    },
+    boolean
+  > = this.commandService.create((payload) => {
+    const [submitSuccess, submitPending, submitFailure] =
+      payload.submitInput.callbackArray;
+    const {
+      paymentMethod,
+      cartId,
+      additionalData,
+      paymentSessionId,
+      returnPath,
+    } = payload.submitInput;
+    const currentDate = new Date();
+    const timeZoneOffset = currentDate.getTimezoneOffset();
+    const submitRequest: SubmitRequest = {
+      paymentMethod,
+      cartId,
+      additionalData,
+      channel: 'BROWSER',
+      browserInfo: {
+        acceptHeader: 'application/json',
+        colorDepth: this.winRef.nativeWindow?.screen?.colorDepth,
+        javaEnabled: false,
+        javaScriptEnabled: true,
+        language: this.winRef.nativeWindow?.navigator?.language,
+        screenHeight: this.winRef.nativeWindow?.screen?.height,
+        screenWidth: this.winRef.nativeWindow?.screen?.width,
+        userAgent: this.winRef.nativeWindow?.navigator?.userAgent,
+        originUrl: this.winRef.nativeWindow?.location?.origin,
+        timeZoneOffset,
+      },
+    };
+    if (paymentMethod !== PaymentMethod.CREDIT_CARD) {
+      submitRequest.encryptedToken = '';
+    }
+
+    return combineLatest([
+      this.userIdService.getUserId(),
+      this.activeCartFacade.getActiveCartId(),
+    ]).pipe(
+      switchMap(([userId, activeCartId]: [string, string]) => {
+        submitRequest.cartId = activeCartId;
+        return this.opfOtpFacade.generateOtpKey(userId, activeCartId);
+      }),
+      filter((response) => Boolean(response?.value)),
+      take(1),
+      concatMap(({ value: otpKey }) =>
+        this.opfPaymentConnector.submitPayment(
+          submitRequest,
+          otpKey,
+          paymentSessionId
+        )
+      ),
+      concatMap((response: SubmitResponse) => {
+        if (
```

Comment:
This response handling also can be a part of the separate method. :) What do you think? I just don't want to overloading the responsibility of `submitPaymentCommand` method.

Reply:
good idea